### DISCUSSION TITLE
Connect using HA Bluetooth proxy

### DISCUSSION THREAD

**Discussion by mlenk** [SBFspot/SBFspot#699] [General]
Hi all,
I'm trying to connect to a SB 5000TL-21, but need to use a Shelly Plus1 as a bluetooth proxy to be in range. I cant't get it to connect to the SB, should a setup like this be possible? Any pointers would be much apprechiated :) Thanks, have nice weekend everyone.
Martin

**Comment by sbf-** [comment]
Try with Sunnyexplorer first. If it works it might be possible to make it work with SBFspot too. I don´t think it will work out of the box with latest version.

**Reply by mlenk** [reply]
I can connect to the SB with Sunnyexplorer, but the PC running that is much closer and has it's own bluetooth dongle. The HomeAssistant machine is much further away. I can connect to other Bluetooth devices through the shelly as a proxy, but don't understand how that works under the hood :)
In the SBFspot config, I tried putting the shelly bt mac address as the host bluetooth mac, and also leaving it empty. Here is the part of the log I believe is relevant:
```
Connecting to 00:80:25:a4:c1:75 (1/10)
Connecting to 00:80:25:a4:c1:75 (2/10)
Connecting to 00:80:25:a4:c1:75 (3/10)
Connecting to 00:80:25:a4:c1:75 (4/10)
Connecting to 00:80:25:a4:c1:75 (5/10)
Connecting to 00:80:25:a4:c1:75 (6/10)
Connecting to 00:80:25:a4:c1:75 (7/10)
Connecting to 00:80:25:a4:c1:75 (8/10)
Connecting to 00:80:25:a4:c1:75 (9/10)
Connecting to 00:80:25:a4:c1:75 (10/10)
Sun May 19 09:13:07 2024: CRITICAL: bthConnect() returned -1
```

Thanks, Martin

**Reply by sbf-** [reply]
Maybe the windows version is an option and send the data to HA with MQTT?

Have you tried `hcitool scan`?

**Reply by mlenk** [reply]
Reading up on how HA Bluetooth proxy works, i came across this: "Shelly devices can only be used to listen (scan) for BT advertisements you can’t control other BT devices using them." Would that be the answer? The SB needs an active bluetooth connection?
How do you mean, the windows version?
Martin

**Reply by sbf-** [reply]
> How do you mean, the windows version?

Run SBFspot on your windows pc (the one with Sunnyexplorer) and configure it to send its data using MQTT to HA
Your PC needs local network access of course. Don't know if that is the case.

**Reply by habuild** [reply]
@mlenk 

It's been awhile since I looked at the HA proxy stuff, I don't think it will work for your use case though. You would need SBFspot running "somewhere" to send the commands to the proxy, then forward those commands to your inverter and then receive the response from the inverter and forward that back to HA and SBFspot. The proxy setup doesn't broadcast commands as far as I am aware. It only receives bluetooth broadcasts. 

The PC forwarding MQTT data to home assistant works. I personally used to have issues with that setup as my PC was bad at power management, but it has an old motherboard. It would forget to wake up and send data if I recall correctly. 

You could try a USB extension and a BT dongle to get closer to your inverter with your HA machine.

or and what I really came to say was..

You could get an esp32 and see if this is working. 

https://github.com/darrylb123/ESP32_SMA-Inverter-MQTT 

or his esphome version
https://github.com/keerekeerweere/esphome_smabluetooth

I never really played around with it much as I didn't need it, but I am fairly sure he had it working. Then you just need to get the MQTT data back to HA via your MQTT broker and wifi.

**Comment by Bastelritius** [comment]
Hi, maybe I can continue this discussion. I also want to try to implement my SMA 5000TL-20 to my HA via a ESP32 bluethoot proxy.
But the problem is, that I can not find the converter in the list of bluethoot devices in HA. I´m sure that the converter is working and it can be connected via bluethooth when I´m trying it with a raspberry, or with windows PC and sunny explorer.
But when I´m using the bluethoot proxy I can not find the converter MAC adress. 
What can be the reason? Can anybody help, or have an idea what I can check. Is it a problem with the Bluethooth proxy? Is there a limitation which devices he is forwarding?

Thanks a lot for all your ideas and comments

**Comment by mlenk** [comment]
6 Month progress report: I gave up, bought a 4€ USB bluetooth dongle and accepted the fact that the HA host has to live in range from the converter :)

**Reply by mgraupner** [reply]
Why not place a Raspberry Pi Zero 2 W close to the inverter which runs SBFspot and pushes the data to your HA/MQTT system?
